x-pack/test_serverless/shared/services/user_manager/saml_auth.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosResponse } from 'axios';
+import Url from 'url';
+import * as cheerio from 'cheerio';
+import { parse as parseCookie } from 'tough-cookie';
+import { createSAMLResponse as createMockedSAMLResponse } from '@kbn/mock-idp-plugin/common';
+import { ToolingLog } from '@kbn/tooling-log';
+import { Session } from './svl_user_manager';
+
+export interface SAMLSessionParams {
+  username: string;
+  password: string;
+  kbnHost: string;
+  kbnVersion: string;
+  log: ToolingLog;
+}
+
+export interface FakeSAMLSessionParams {
+  username: string;
+  email: string;
+  fullname: string;
+  role: string;
+  kbnHost: string;
+  log: ToolingLog;
+}
+
+const envHosts: { [key: string]: string } = {
+  qa: 'console.qa.cld.elstc.co',
+  staging: 'staging.found.no',
+  production: 'api.elastic-cloud.com',
```

Comment:
is it safe to hardcode all our envs in github?

Reply:
Thanks for pointing out, waiting for Infosec feedback

Reply:
removed 8ad1b7e1f672cf4d111ae0662fe54e8dd4aedbd0